Bearded.UI/Events/EventRouter.cs

```diff
@@ -42,12 +36,60 @@ public static EventPropagationPath FindPropagationPath(IControlParent root, Cont
                     return EventPropagationPath.Empty;
                 }
                 path.Add(parentAsControl);
-                
+
                 parent = parentAsControl.Parent;
             }
 
             path.Reverse();
             return new EventPropagationPath(path.AsReadOnly());
         }
+
+        private static IEnumerable<Control> findPropagationPathUsingTest(
+            IControlParent root, PropagationTest propagationTest)
+        {
+            var path = new List<Control>();
```

Comment:
Don't you need to add the root to the path? And potentially catch the root == null case?

Reply:
No actually. If the root isn't being hit by the propagation test, we should return an empty propagation path.

Reply:
But you never check the root at all, and it will never be added.

Reply:
This is WAI, as root control is assumed to not be a control. Added comments to make this explicit though.